Visitor/Arithmetic.php

```diff
@@ -245,18 +246,39 @@ public function visit(
 
             case 'token':
                 $value = $element->getValueValue();
+                $token = $element->getValueToken();
                 $out   = null;
 
-                if ('constant' === $element->getValueToken()) {
+                if ('constant' === $token) {
                     if (defined($value)) {
                         $out = constant($value);
                     } else {
                         $out = $this->getConstant($value);
                     }
-                } elseif ('id' === $element->getValueToken()) {
+                } elseif ('id' === $token) {
                     return $value;
                 } else {
-                    $out = (float) $value;
+                    $number = preg_replace('/^0[xXbB]/', '', $value);
+
+                    switch ($token) {
+                        case 'binary':
+                            $number = bindec($number);
+                            break;
+
+                        case 'hexadecimal':
+                            $number = hexdec($number);
+                            break;
+
+                        case 'octal':
+                            $number = octdec($number);
+                            break;
+                    }
+
+                    if ('hexadecimal' === $token && $number > PHP_INT_MAX) {
+                        throw new OutOfBounds('Number ' . $value . ' is out of bound for the ' . php_uname('m') . ' platform.');
```

Comment:
Actually, this is not totally correct. We can have a double/real written in the hexa or octal notation. Do we restrict binary, hexadecimal and octal notation to integers only? If yes, this is still not correct in case of an integer overflow: You must check it is greater or equal to `0`.